specifyweb/frontend/js_src/lib/components/Statistics/StatItems.tsx

```diff
@@ -0,0 +1,255 @@
+import type { SpQueryField, Tables } from '../DataModel/types';
+import type { RA } from '../../utils/types';
+import type { SerializedResource } from '../DataModel/helperTypes';
+import { StatsResult } from './StatsResult';
+import React from 'react';
+import {
+  queryCountPromiseGenerator,
+  useCustomStatsSpec,
+  useResolvedSpec,
+  useResolvedSpecToQueryResource,
+} from './hooks';
+import type { CustomStat, DefaultStat, StatsSpec } from './types';
+import { SpecifyResource } from '../DataModel/legacyTypes';
+import { SpQuery } from '../DataModel/types';
+import { useAsyncState } from '../../hooks/useAsyncState';
+import { throttledAjax } from '../../utils/ajax/throttledAjax';
+import { BackendStatsResult } from './types';
+import { ajax } from '../../utils/ajax';
+
+export function StatItem({
+  statsSpec,
+  item,
+  categoryIndex,
+  itemIndex,
+  onRemove: handleRemove,
+  onClick: handleClick,
+  onSpecChanged: handleSpecChanged,
+  onValueLoad: handleValueLoad,
+  onItemRename: handleItemRename,
+}: {
+  readonly statsSpec: StatsSpec;
+  readonly item: CustomStat | DefaultStat;
+  readonly categoryIndex: number;
+  readonly itemIndex: number;
+  readonly onRemove: (() => void) | undefined;
+  readonly onClick: (() => void) | undefined;
+  readonly onSpecChanged:
+    | ((
+        tableName: keyof Tables,
+        fields: RA<
+          Partial<SerializedResource<SpQueryField>> & { readonly path: string }
+        >,
+        itemName: string
+      ) => void)
+    | undefined;
+  readonly onValueLoad:
+    | ((
+        categoryIndex: number,
+        itemIndex: number,
+        value: number | string,
+        itemLabel: string
+      ) => void)
+    | undefined;
+  readonly onItemRename: ((newLabel: string) => void) | undefined;
+}): JSX.Element | null {
+  const handleItemValueLoad = React.useCallback(
```

Comment:
You have access to itemLabel in this component, so don't need QueryItem/BackEndItem provide it again